doc/ug/io.tex

```diff
@@ -1217,6 +1217,256 @@ \section{Error handling}
+\keyword{list}: A list of particle radii, indexed by the particle type.
+\item[\opt{particle_coloring}] 
+\keyword{"auto"} (default): Colors of charged particles are specified by 
+\keyword{particle_charge_colors}, neutral particles by \keyword{particle_type_colors}\\
+\keyword{"charge"}: Minimum and maximum charge of all particles is determined by the visualizer.
+All particles are colored by a linear interpolation of the two colors given by 
+\keyword{particle_charge_colors} according to their charge.\\
+\keyword{"type"}: Particle colors are specified by \keyword{particle_type_colors}, 
+indexed by their numerical particle type.
+\item[\opt{particle_type_colors}] Colors for particle types.
+\item[\opt{particle_type_materials}] Materials of the particle types.
+\item[\opt{particle_charge_colors}] Two colors for min/max charged particles.
+
+\item[\opt{draw_constraints}] Enables constraint visualization. For simple constraints.
+(planes, spheres and cylinders), OpenGL primitives are used. Otherwise, visualization by rasterization is used.
+ \item[\opt{rasterize_pointsize}] Point size for the rasterization dots.
+\item[\opt{rasterize_resolution}] Accuracy of the rasterization.
+\item[\opt{quality_constraints}] The number of subdivisions for primitive constraints.
+\item[\opt{constraint_type_colors}] Colors of the constaints by type.
+\item[\opt{constraint_type_materials}] Materials of the constraints by type.
+
+\item[\opt{draw_bonds}] Enables bond visualization.
+\item[\opt{bond_type_radius}] Radii of bonds by type.
+\item[\opt{bond_type_colors}] Color of bonds by type.
+\item[\opt{bond_type_materials}] Materials of bonds by type.
+
+\item[\opt{ext_force_arrows}] Enables external force visualization.
+\item[\opt{ext_force_arrows_scale}] Scale factor for external force arrows.
+\item[\opt{drag_enabled}] Enables mouse-controlled particles dragging (Default: False)
+\item[\opt{drag_force}] Factor for particle dragging 
+
+\item[\opt{light_pos}] If \keyword{"auto"} (default) is used, the light is placed dynamically
+in the particle barycenter of the system. Otherwise, a fixed coordinate can be set.
+\item[\opt{light_color}] Light color
+\item[\opt{light_brightness}] Brightness (inverse constant attenuation) of the light.
+\item[\opt{light_size}] Size (inverse linear attenuation) of the light.
+
+\end{arguments}
+
+The optional parameters to adjust the appearance of the visualization
+have suitable default values for most simulations. Colors for particles, bonds
+and constraints are specified by RGBA arrays, materials by an array for the
+ambient, diffuse and specular (ADS) components. To distinguish particle
+groups, arrays of RGBA or ADS entries are used, which are indexed circularly by
+the numerical particle type.
+
+\subsubsection{Keyboard controls}
+
+The camera is controlled via mouse (camera direction), WASD-Keyboard control
+(camera movement) and the key pairs QE, RF, ZC (camera roll).
```

Comment:
Why QE, RF, ZC? Better use something that's together pairwise, like IJ, OK, PL. Also, how does WASD work in three dimensions?